src/views/Correction.vue

```diff
@@ -152,15 +202,28 @@ export default class Correction extends Mixins(DateMixin, FilingTemplateMixin, L
         // parse correction filing into store
         // this applies the diffs (corrections)
         this.parseCorrection(correctionFiling)
-      }
-
-      if (this.correctedId) {
+      } else if (this.correctedId) {
         // fetch original IA to correct
         this.correctedFiling = await this.fetchFilingById(this.correctedId)
 
+        // do not proceed if this isn't an IA filing
+        if (!this.correctedFiling.incorporationApplication) {
+          console.log('Invalid IA filing')
```

Comment:
The warning below is pretty darn cool. I know some projects are strict on the no console logs.. personally i don't mind them.

Reply:
Hah, this code is redundant anyway! In this "if" block I first log the error, then throw the same message; then the catch block logs the error again before handling it.
Will fix...

Reply:
Updated.